packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -245,6 +246,37 @@ export class Pipeline extends cdk.Construct implements events.IEventRuleTarget {
     this.stages.splice(index, 0, stage);
   }
 
+  // ignore unused private method (it's actually used in Stage)
+  // @ts-ignore
+  private _generateOutputArtifactName(stage: actions.IStage, action: actions.Action): string {
+    // for now, just return a generic output artifact,
+    // ignoring the names of both the Stage, and the Action
+    return 'Artifact_' + (++this.artifactsCounter);
+  }
+
+  // ignore unused private method (it's actually used in Stage)
+  // @ts-ignore
+  private _findInputArtifactFor(stage: actions.IStage, action: actions.Action): actions.Artifact {
```

Comment:
The defaulting behavior that involves this particular element has a caveat that changing the `runOrder` after the defaulting behavior has happened will cause incorrect values to be used... A solutions is to make the `runOder` immutable & set at construction time?

Reply:
Yep. Not sure if it's a huge deal, as you can always name the artifacts directly, but I'm fine with making `runOrder` immutable (the rest of the properties of Actions already are).